packages/core/src/errors.ts

```diff
@@ -35,3 +35,36 @@ export class SessionError extends RetryRequestError {
         super(`Detected a session error, rotating session... ${message ? `\n${message}` : ''}`);
     }
 }
+
+export class ContextPipelineInterruptedError extends Error {
+    constructor(message?: string) {
+        super(`Request handling was interrupted during context initialization ${message ? ` - ${message}` : ''}`);
+    }
+}
+
+export class ContextPipelineInitializationError extends Error {
+    constructor(
+        public error: unknown,
+        public crawlingContext: {},
```

Comment:
```typescript 
try {}
catch (e: ContextPipelineInitializationError) {
    console.error(e);
}
```

Logging the error like this will log the entire crawling context into the console. This can be quite heavy (especially if it happens repeatedly). 

I see the context being useful for debugging as well, so this is rather a question - are you aware / sure about this?

Reply:
I'm aware of that, it actually bit me earlier :grin: After some consideration, we probably don't need the context here. The Python version does, but here we can make do with a simple cast.